.github/workflows/linux.yml

```diff
@@ -0,0 +1,45 @@
+name: linux
+
+on: push
+
+env:
+  RUSTDOCFLAGS: "-D warnings"
+  TOOLCHAIN: nightly
+  CARGO_TERM_COLOR: always
+
+jobs:
+  docs:
+    # rust cross-compilation
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.TOOLCHAIN }}
+          default: true
+      - name: Cache Cargo
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin
+            ~/.cargo/git
+            ~/.cargo/registry
+          key: ${{ github.workflow }}-${{ github.job }}-toolchain-${{ env.TOOLCHAIN }}-cargo-${{ env.CARGO_NDK_REV }}-${{ env.CARGO_TAI_REV }}-${{ hashFiles('**/Cargo.lock') }}
+      - name: Cache target
+        id: cache-target
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-target
+        with:
+          path: target
+          key: ${{ github.workflow }}-${{ github.job }}-${{ env.cache-name }}-${{ env.TOOLCHAIN }}-cargo-${{ env.CARGO_NDK_REV }}-${{ env.CARGO_TAI_REV }}-${{ hashFiles('**/Cargo.lock') }}
+      - name: Build phoenix_channels_client
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
```

Comment:
I'm not sure if `build` checks that the rust in the docs is correct. That might be done via `cargo test` (which I see requires a server). Maybe just `build --all-targets` (I'm not sure the notation with `actions-rs/cargo`).

Reply:
`cargo build --all-targets` does not catch errors in the doctests. I tried putting a typo in the doctests for the `socket` module and it wasn't taught until `cargo test`.

Reply:
Added `--all-targets` anyway.